browser/file_persister.go

```diff
@@ -0,0 +1,106 @@
+package browser
+
+import (
+	"errors"
+	"fmt"
+	"net/url"
+	"strings"
+
+	"github.com/grafana/xk6-browser/env"
+	"github.com/grafana/xk6-browser/storage"
+)
+
+type presignedURLConfig struct {
+	getterURL string
+	headers   map[string]string
+	basePath  string
+}
+
+// newFilePersister will return either a persister that persists file to the local
+// disk or uploads the files to a remote location. This decision depends on whether
+// the K6_BROWSER_SCREENSHOTS_OUTPUT env var is setup with the correct configs.
+func newFilePersister(envLookup env.LookupFunc) (filePersister, error) {
```

Comment:
```
// newScreenshotPersister will return either a persister that persists file to the local
// disk or uploads the files to a remote location. This decision depends on whether
// the K6_BROWSER_SCREENSHOTS_OUTPUT env var is setup with the correct configs.
func newScreenshotPersister(envLookup env.LookupFunc) (filePersister, error) {
```

I know you wanted to keep it generic, but it uses `ScreenshotsOutput`. So it makes better sense to me to rename it like this to reflect its purpose.

Reply:
Fixed in ab27742ff7b334e8db66668d1fb2fdaa531d5a10